src/workerd/jsg/wrappable.h

```diff
@@ -47,29 +47,16 @@ class HeapTracer;
 // Wrappable and are not visible to GC tracing.
 class Wrappable: public kj::Refcounted {
 public:
-  // Number of internal fields in a wrapper object.
-  static constexpr uint INTERNAL_FIELD_COUNT = 3;
 
-  // Index of the internal field that points back to the `Wrappable`.
-  static constexpr uint WRAPPED_OBJECT_FIELD_INDEX = 2;
+  enum InternalFields {
+    // Index of the internal field that points back to the `Wrappable`.
+    WRAPPED_OBJECT_FIELD_INDEX,
+    // Number of internal fields in a wrapper object.
+    INTERNAL_FIELD_COUNT,
+  };
 
-  // Index of the internal field that contains a pointer to the cppgc shim object, used to get
-  // tracing callback from V8 and get notification when the wrapper is collected.
-  static constexpr uint CPPGC_SHIM_FIELD_INDEX = 1;
-
-  // Field must contain a pointer to `WRAPPABLE_TAG`. This is a requirement of v8::CppHeap. The
-  // purpose is not clear.
-  //
-  // Note that although V8 lets us configure which slot this tag is in, in practice if we set it
-  // to anything other than zero, we see crashes inside V8. It appears that V8 allocates some
-  // objects of its own which have internal fields, and then GC doesn't check that the index is
-  // in-bounds before reading it...
-  static constexpr uint WRAPPABLE_TAG_FIELD_INDEX = 0;
-
-  // The value pointed to by the internal field field `WRAPPABLE_TAG_FIELD_INDEX`.
-  //
-  // This value was chosen randomly.
-  static constexpr uint16_t WRAPPABLE_TAG = 0xeb04;
+  static constexpr v8::CppHeapPointerTag WRAPPABLE_TAG =
+      v8::CppHeapPointerTag::kDefaultTag;
```

Comment:
We should probably have our own tag so that we can check if objects are our wrappables (see other comment).

Reply:
Doing so would require a patch to v8 to modify the `CppHeapPointerTag` enum class. Certainly possible to do but I was rather hoping to avoid adding another floated patch

Reply:
Oh, I see the internal enum `v8::internal::ExternalPointerTag` actually has quite a few different values and the integer values actually encode structured information in the bits... huh...

So obviously we _can't_ actually extend this enum.